### PR TITLE
[5.6] Fix cache for loggers

### PR DESCRIPTION
**Pull request by CyrilMazur**
This fixes https://github.com/laravel/framework/issues/23117

### CODE REVIEW

**Comment by laurencei** [comment]
Wouldnt a better option be to register this as a singleton - and let the container handle it?

**Comment by laurencei** [comment]
Could you add some tests to show it is getting the same instance? That will help prevent regressions later.

**Comment by CyrilMazur** [comment]
@laurencei done

File: tests/Log/LogManagerTest.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class LogManagerTest extends TestCase
+{
+    public function testLogManagerCachesLoggerInstances()
+    {
+        $manager = new LogManager($this->app);
+
+        $logger1 = $manager->channel('single')->getLogger();
+        $logger2 = $manager->channel('single')->getLogger();
+
+        $this->assertEquals(spl_object_id($logger1), spl_object_id($logger2));
```
Comment (sisve) [inline tests/Log/LogManagerTest.php]: This could probably be an $this->assertSame($logger1, $logger2);

Comment (GrahamCampbell) [reply tests/Log/LogManagerTest.php]: Just do assertSame on the objects.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by CyrilMazur** [comment]
@GrahamCampbell done. I realise I opened this PR into the wrong branch. I intended this PR for 5.6 (it's a bug fix), should I open a new PR?

**Comment by GrahamCampbell** [comment]
Hard reset your branch to 5.6, then cherry-pick your commits and force push.

**Comment by CyrilMazur** [comment]
Thanks. Done.

**Pull request merged**